packages/labs/compiler/src/lib/template-transform.ts

```diff
@@ -261,6 +261,46 @@ class CompiledTemplatePass {
           templateExpression.head.text,
           ...templateExpression.templateSpans.map((s) => s.literal.text),
         ] as unknown as TemplateStringsArray);
+
+    // Just for the deprecated octal check, we need to use `rawText`. Otherwise
+    // strings are escaped in strange ways which cause the regex to fail. This
+    // check is needed because an invalid octal sequence in an uncompiled
+    // template is detected by our `html` tag function. But when compiled, the
+    // invalid octal sequence creates something like the following:
+    //
+    //   brand`\2`
+    //
+    // This results in an undefined entry in the template strings array. lit
+    // then simply renders the `undefined` directly on the page as text. A worse
+    // result is if the octal is adjacent to other content, e.g.
+    // html`<p>Hi</p>\2`. If compiled naively, this will only render `undefined`
+    // on the page. Thus we do not compile templates containing invalid octal
+    // sequences.
+    //
+    // Reference:
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Deprecated_octal
+    const rawTextForOctalCheck = ts.isNoSubstitutionTemplateLiteral(
+      templateExpression
+    )
+      ? ([templateExpression.rawText] as unknown as TemplateStringsArray)
+      : ([
+          templateExpression.head.rawText,
+          ...templateExpression.templateSpans.map((s) => s.literal.rawText),
+        ] as unknown as TemplateStringsArray);
+
+    for (const staticString of rawTextForOctalCheck) {
+      // Regex to detect if an invalid octal sequence was used. It also detects
+      // the invalid sequences: `\8` and `\9`.
+      // References:
+      //   - eslint no-octal-escape rule:
+      //     https://github.com/eslint/eslint/blob/main/lib/rules/no-octal-escape.js
+      //   - Chromium octal escape sequence scanner:
+      //     https://source.chromium.org/chromium/chromium/src/+/181fa1f62f501c27ed19fcb69206a0e2e1eff513:v8/src/parsing/scanner.cc;l=452-460
+      if (/^([^\\|\s]|\\.)*?\\(0)*[1-9]/gm.test(staticString)) {
```

Comment:
I often prefer to put regexes into named variables to be self-descriptive and have a nice place to put comments like these.

Reply:
Another great style nit! Applied 😄